analytics/analytics/job_processor/dimensions.py

```diff
@@ -277,16 +277,26 @@ def create_package_spec_dimension(info: PackageInfo | None) -> PackageSpecDimens
     if info is None:
         return PackageSpecDimension.get_empty_row()
 
-    spec, _ = PackageSpecDimension.objects.get_or_create(
-        hash=info.hash,
-        defaults={
-            "name": info.name,
-            "version": info.version,
-            "compiler_name": info.compiler_name,
-            "compiler_version": info.compiler_version,
-            "arch": info.arch,
-            "variants": info.variants,
-        },
-    )
-
-    return spec
+    existing_spec = PackageSpecDimension.objects.filter(hash=info.hash).first()
+
+    if existing_spec is None:
+        return PackageSpecDimension.objects.create(
+            hash=info.hash,
+            name=info.name,
+            version=info.version,
+            compiler_name=info.compiler_name,
+            compiler_version=info.compiler_version,
+            arch=info.arch,
+            variants=info.variants,
+        )
+
+    changes_made = False
+    for field in ["name", "version", "compiler_name", "compiler_version", "arch", "variants"]:
```

Comment:
This will avoid needing to manually specify the fields
```suggestion
    for field in [field.name for field in PackageSpecDimension._meta.fields]:
```